### PR TITLE
[WIP] Mutation observer test fix

### LINKED ISSUE #21233: Investigate and fix remaining MutationObserver test failures
https://github.com/servo/servo/blob/master/tests/wpt/metadata/dom/nodes/MutationObserver-attributes.html.ini
https://github.com/servo/servo/blob/master/tests/wpt/metadata/dom/nodes/MutationObserver-characterData.html.ini
https://github.com/servo/servo/blob/master/tests/wpt/metadata/dom/nodes/MutationObserver-document.html.ini

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #21233.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21456)

### CODE REVIEW

**Comment by KiChjang** [comment]
Ok, so I'm quite convinced that this is a matter of ordering. During debugging, I'm able to see that mutation events are being fired for adding `inserted_element` and `inserted_script`, but they are ordered AFTER the text node additions for `<div id=log>`:

<details>

```
pid:55426 MutationRecord {
 0:48.85 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.85 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 7, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.85 pid:55426  added: Some([NodeInfo { uniqueId: "1e9985eb5dc24cb0aae52d53b9b02320", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "P", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "n00" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.85 pid:55426  removed: None,
 0:48.85 pid:55426  prev: Some(NodeInfo { uniqueId: "d45e837b2783431da87ee94071d7889c", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s001" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.85 pid:55426  next: None,
 0:48.85 pid:55426 }
 0:48.85 pid:55426 MutationRecord {
 0:48.85 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 7, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "b0ee85ddec58463facd61eba1ba3ad04", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s002" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.86 pid:55426  removed: None,
 0:48.86 pid:55426  prev: Some(NodeInfo { uniqueId: "1e9985eb5dc24cb0aae52d53b9b02320", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "P", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "n00" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "b0ee85ddec58463facd61eba1ba3ad04", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s002" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "5fd5516fe36c4a08b127a50d276e092e", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "b0ee85ddec58463facd61eba1ba3ad04", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "\n  var newScript = document.createElement(\"script\");\n  setupTest.step(function() {\n    newScript.textContent = \"document.body.appendChild(newElement);\";\n    newScript.id = \"inserted_script\";\n    document.getElementById(\"n00\").appendChild(newScript);\n  });\n  if (testCounter < 1) {\n    insertionTest.step(\n      function () {\n        assert_unreached(\"document observer did not trigger\");\n      });\n  }\n", incompleteValue: false }]),
 0:48.86 pid:55426  removed: None,
 0:48.86 pid:55426  prev: None,
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "2fedba3195df444c9af889e61e522797", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "a127cf5cd200497c83e751f23e731663", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 0 complete, 3 remain", incompleteValue: false }]),
 0:48.86 pid:55426  removed: Some([NodeInfo { uniqueId: "dff4663f57134b52b9dc8913b23eac67", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 0 complete, 3 remain", incompleteValue: false }]),
 0:48.86 pid:55426  prev: None,
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "a92d7c0a78b5498889056fef3a591802", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 0 complete, 3 remain", incompleteValue: false }]),
 0:48.86 pid:55426  removed: Some([NodeInfo { uniqueId: "2fedba3195df444c9af889e61e522797", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 0 complete, 3 remain", incompleteValue: false }]),
 0:48.86 pid:55426  prev: None,
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "989393d516004380b95d357248cc6864", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 1 complete, 2 remain", incompleteValue: false }]),
 0:48.86 pid:55426  removed: Some([NodeInfo { uniqueId: "a92d7c0a78b5498889056fef3a591802", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 0 complete, 3 remain", incompleteValue: false }]),
 0:48.86 pid:55426  prev: None,
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.86 pid:55426  added: Some([NodeInfo { uniqueId: "68db8eca1069445c94c358231841b748", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "a127cf5cd200497c83e751f23e731663", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 1 complete, 2 remain", incompleteValue: false }]),
 0:48.86 pid:55426  removed: Some([NodeInfo { uniqueId: "989393d516004380b95d357248cc6864", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 1 complete, 2 remain", incompleteValue: false }]),
 0:48.86 pid:55426  prev: None,
 0:48.86 pid:55426  next: None,
 0:48.86 pid:55426 }
 0:48.86 pid:55426 MutationRecord {
 0:48.86 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.86 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.87 pid:55426  added: Some([NodeInfo { uniqueId: "7dea6a9fe3124eeb9121288a5be03b72", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "a127cf5cd200497c83e751f23e731663", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 1 complete, 2 remain", incompleteValue: false }]),
 0:48.87 pid:55426  removed: Some([NodeInfo { uniqueId: "68db8eca1069445c94c358231841b748", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 1 complete, 2 remain", incompleteValue: false }]),
 0:48.87 pid:55426  prev: None,
 0:48.87 pid:55426  next: None,
 0:48.87 pid:55426 }
 0:48.87 pid:55426 MutationRecord {
 0:48.87 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.87 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 8, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.87 pid:55426  added: Some([NodeInfo { uniqueId: "4f99618b97f84e5baa47f31714240a1c", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SPAN", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "inserted_element" }, AttrInfo { namespace: "", name: "style", value: "display: none" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.87 pid:55426  removed: None,
 0:48.87 pid:55426  prev: Some(NodeInfo { uniqueId: "b0ee85ddec58463facd61eba1ba3ad04", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s002" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.87 pid:55426  next: None,
 0:48.87 pid:55426 }
 0:48.87 pid:55426 MutationRecord {
 0:48.87 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.87 pid:55426  target: NodeInfo { uniqueId: "1e9985eb5dc24cb0aae52d53b9b02320", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "P", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "n00" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.87 pid:55426  added: Some([NodeInfo { uniqueId: "9708b29cebdc47a48cfc60f733388dff", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "1e9985eb5dc24cb0aae52d53b9b02320", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "inserted_script" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.87 pid:55426  removed: None,
 0:48.87 pid:55426  prev: None,
 0:48.87 pid:55426  next: None,
 0:48.87 pid:55426 }
 0:48.87 pid:55426 MutationRecord {
 0:48.87 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.87 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 9, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.87 pid:55426  added: Some([NodeInfo { uniqueId: "b0e48f608abf40e4816695130ba2c8f9", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s003" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.87 pid:55426  removed: None,
 0:48.87 pid:55426  prev: Some(NodeInfo { uniqueId: "4f99618b97f84e5baa47f31714240a1c", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SPAN", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "inserted_element" }, AttrInfo { namespace: "", name: "style", value: "display: none" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.87 pid:55426  next: None,
 0:48.87 pid:55426 }
 0:48.87 pid:55426 MutationRecord {
 0:48.87 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.87 pid:55426  target: NodeInfo { uniqueId: "b0e48f608abf40e4816695130ba2c8f9", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s003" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.87 pid:55426  added: Some([NodeInfo { uniqueId: "13f838fcca714670987393e80c6617c3", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "b0e48f608abf40e4816695130ba2c8f9", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "\n  setupTest.step(function() {\n    document_observer.disconnect();\n  });\n  if (testCounter < 2) {\n    insertionTest2.step(\n      function () {\n        assert_unreached(\"document observer did not trigger\");\n      });\n  }\n  insertionTest.done();\n  insertionTest2.done();\n", incompleteValue: false }]),
 0:48.87 pid:55426  removed: None,
 0:48.87 pid:55426  prev: None,
 0:48.88 pid:55426  next: None,
 0:48.88 pid:55426 }
 0:48.89 pid:55426 MutationRecord {
 0:48.89 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.89 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 11, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.89 pid:55426  added: None,
 0:48.89 pid:55426  removed: Some([NodeInfo { uniqueId: "e7e336cc6c4e4e54b48419917d2e0d94", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 2, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "d01" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.89 pid:55426  prev: Some(NodeInfo { uniqueId: "2571d7673190471098f52f135d77f73d", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "P", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "n012" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.89 pid:55426  next: None,
 0:48.89 pid:55426 }
 0:48.89 pid:55426 MutationRecord {
 0:48.89 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.89 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.89 pid:55426  added: Some([NodeInfo { uniqueId: "52a48d3a42b844b29c77ab55cb62d366", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "a127cf5cd200497c83e751f23e731663", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 2 complete, 2 remain", incompleteValue: false }]),
 0:48.89 pid:55426  removed: Some([NodeInfo { uniqueId: "4d32d16a3adc4fc58e16a2b763815bac", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 2 complete, 2 remain", incompleteValue: false }]),
 0:48.89 pid:55426  prev: None,
 0:48.89 pid:55426  next: None,
 0:48.89 pid:55426 }
 0:48.89 pid:55426 MutationRecord {
 0:48.89 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.89 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.89 pid:55426  added: Some([NodeInfo { uniqueId: "bcdc55c0fc0f4882961ca39d4f1d3f93", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 2 complete, 2 remain", incompleteValue: false }]),
 0:48.89 pid:55426  removed: Some([NodeInfo { uniqueId: "52a48d3a42b844b29c77ab55cb62d366", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 2 complete, 2 remain", incompleteValue: false }]),
 0:48.89 pid:55426  prev: None,
 0:48.89 pid:55426  next: None,
 0:48.89 pid:55426 }
 0:48.89 pid:55426 MutationRecord {
 0:48.89 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.90 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.90 pid:55426  added: Some([NodeInfo { uniqueId: "d5c1b62d9858401db8ed70d8c3ab779f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 3 complete, 1 remain", incompleteValue: false }]),
 0:48.90 pid:55426  removed: Some([NodeInfo { uniqueId: "bcdc55c0fc0f4882961ca39d4f1d3f93", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 2 complete, 2 remain", incompleteValue: false }]),
 0:48.90 pid:55426  prev: None,
 0:48.90 pid:55426  next: None,
 0:48.90 pid:55426 }
 0:48.90 pid:55426 MutationRecord {
 0:48.90 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.90 pid:55426  target: NodeInfo { uniqueId: "a127cf5cd200497c83e751f23e731663", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "DIV", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "log" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.90 pid:55426  added: Some([NodeInfo { uniqueId: "7c3c7078f159441b9b7616f01484b63e", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "a127cf5cd200497c83e751f23e731663", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 3 complete, 1 remain", incompleteValue: false }]),
 0:48.90 pid:55426  removed: Some([NodeInfo { uniqueId: "d5c1b62d9858401db8ed70d8c3ab779f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "Running, 3 complete, 1 remain", incompleteValue: false }]),
 0:48.90 pid:55426  prev: None,
 0:48.90 pid:55426  next: None,
 0:48.90 pid:55426 }
 0:48.90 pid:55426 MutationRecord {
 0:48.90 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.90 pid:55426  target: NodeInfo { uniqueId: "e9aa2a2d4d0b41cc9c01029372630a2f", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9d4fe44c7674fc58afa8bba09319413", nodeType: 1, namespaceURI: "", nodeName: "BODY", numChildren: 12, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.90 pid:55426  added: Some([NodeInfo { uniqueId: "c9d0264a9f8240639f7d7bf4918df8f6", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s012" }], isDocumentElement: false, shortValue: "", incompleteValue: false }]),
 0:48.90 pid:55426  removed: None,
 0:48.90 pid:55426  prev: Some(NodeInfo { uniqueId: "2571d7673190471098f52f135d77f73d", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "P", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "n012" }], isDocumentElement: false, shortValue: "", incompleteValue: false }),
 0:48.90 pid:55426  next: None,
 0:48.90 pid:55426 }
 0:48.90 pid:55426 MutationRecord {
 0:48.90 pid:55426  record_type: DOMString("childList", PhantomData),
 0:48.90 pid:55426  target: NodeInfo { uniqueId: "c9d0264a9f8240639f7d7bf4918df8f6", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "e9aa2a2d4d0b41cc9c01029372630a2f", nodeType: 1, namespaceURI: "", nodeName: "SCRIPT", numChildren: 1, name: "", publicId: "", systemId: "", attrs: [AttrInfo { namespace: "", name: "id", value: "s012" }], isDocumentElement: false, shortValue: "", incompleteValue: false },
 0:48.90 pid:55426  added: Some([NodeInfo { uniqueId: "351011d49f6a4008a134e552677aff22", baseURI: "http://web-platform.test:8000/dom/nodes/MutationObserver-document.html", parent: "c9d0264a9f8240639f7d7bf4918df8f6", nodeType: 3, namespaceURI: "", nodeName: "#text", numChildren: 0, name: "", publicId: "", systemId: "", attrs: [], isDocumentElement: false, shortValue: "\n  setupTest.step(function() {\n    document2_observer.disconnect();\n  });\n  if (testCounter < 2) {\n    removalTest.step(\n      function () {\n        assert_unreached(\"document observer did not trigger\");\n      });\n  }\n  removalTest.done();\n  setupTest.done();\n", incompleteValue: false }]),
 0:48.90 pid:55426  removed: None,
 0:48.90 pid:55426  prev: None,
 0:48.90 pid:55426  next: None,
 0:48.90 pid:55426 }
```

</details>

File: components/script/dom/globalscope.rs [hunk 1]
```diff
@@ -1701,7 +1701,14 @@ impl GlobalScope {
                 maybe_resume_unwind();
                 result
             },
-        )
+        );
+
+        // Step 9
```
Comment (asajeffrey) [inline components/script/dom/globalscope.rs]: Step 9 of which spec?

**Review by asajeffrey — APPROVED** [review]
A small nit, otherwise LGTM.

**Comment by KiChjang** [comment]
Solved by #25515.

**Pull request closed without merging**